src/java.desktop/macosx/native/libosxapp/ThreadUtilities.m

```diff
@@ -59,12 +60,20 @@ + (void)initialize {
                                            nil];
 }
 
++ (void)setApplicationOwner:(BOOL)owner {
+    isNSApplicationOwner = owner;
+}
+
 + (JNIEnv*)getJNIEnv {
```

Comment:
The cached version of 'getJNIEnv' was added for performance reasons to skip `attachCurrentThread` if we know for sure that initialization has already completed. But is it still necessary to track all this flags now, maybe the `attachCurrentThread` now smart enough to check that the current thread was already attached, or it is just fast enough?

Reply:
It seems best to make minimal changes in the normal case where AWT creates the toolkit, which this PR does.

You raise a good question as to whether we could get rid of the caching entirely. That seems like something to consider in a follow-up. We would need to do measure the performance impact.